source/LifecycleMethod.h

```diff
@@ -93,6 +93,33 @@ class LifecycleMethodCall {
   std::optional<std::string> defined_in_derived_class_;
 };
 
+  class LifeCycleMethodGraph {
+    public:
+      LifeCycleMethodGraph() : entry_point_("", "", {}, std::nullopt) {}
+      void addNode(const LifecycleMethodCall& node);
+      void addEdge(const LifecycleMethodCall& from, const LifecycleMethodCall& to);
+      const std::vector<LifecycleMethodCall>& getNeighbours(const LifecycleMethodCall& node) const;
+      bool operator==(const LifeCycleMethodGraph& other) const;
+
+      INCLUDE_DEFAULT_COPY_CONSTRUCTORS_AND_ASSIGNMENTS(LifeCycleMethodGraph)
+
+      static LifeCycleMethodGraph from_json(const Json::Value& value);
+
+    private:
+      struct NodeHasher {
+        std::size_t operator()(const LifecycleMethodCall& node) const {
+          return std::hash<std::string>{}(node.to_string());
+        }
+      };
+
+      std::unordered_map<LifecycleMethodCall, std::vector<LifecycleMethodCall>, NodeHasher> adj_list_;
```

Comment:
This representation doesn't seem quite right. Why is the key a MethodCall itself? I thought we said it should be the name used as the key in the json. It should just be a string.
The parsing logic is quite weird, since we seem to create one node for each call, but a node is a `std::vector`. Basically, you are never using vectors of more than 1 element.

I think this should be redesigned to be something like:
`std::unordered_map<std::stringg, LifecycleGraphNode> nodes_`
And `LifecycleGraphNode` should have a `std::vector<LifecycleMethodCall>` has well as a `std::vector<std::string> successors`.